clang/test/Modules/invalid-module-dep.c

```diff
@@ -0,0 +1,62 @@
+/// This tests the expected error case when there is a mismatch between the pcm dependencies passed in 
+/// the command line with `fmodule-file` and whats encoded in the pcm. 
+
+/// The steps are: 
+/// 1. Build module (A-1) with no dependencies.
+/// 2. Build the same module with files that resolve from different search paths. (A-2).
+/// 3. Build module (B) that depends on the earlier module (A-1).
+/// 4. Build client that depends on both modules (A & B) but depends on a different version (A-2).
```

Comment:
I didn't understand this description initially, only seeing the actual commands helped me. Up to you how much you want to invest into the description but for me the main problem is that it's unclear what is a module name and what is a file name.

Reply:
Updated, let me know if it still reads confusingly.

Reply:
It is great now. And it is way better than what I had in mind. Thanks for improving the description!